apache/superset
# chore(explore): Migrate BigNumber to v1 api [ID-28][ID-55]

### SUMMARY
This PR's goal is to make the legacy Big Number chart plugin not-so-legacy-anymore.

First improvement is migrating the chart endpoint from legacy to V1 API. This change will also allow proper handling of timestamp format metrics. The legacy API assumed that metric result is always a number - if you used a metric that should return a datetime (e.g. `MAX(temporal_column)`) with a db that returns dates as strings (like sqlite or druid), you'd get nulls in response (see screenshots).

Second thing - to render the trendline, we used `@data-ui/xy-chart` library, which hasn't been maintained for 2 years. I replaced it with Echarts, which is our go-to viz library.

Lastly, the legacy plugin wasn't able to handle rolling aggregates properly if values were missing. The new version ensures that rolling windows are resampled with null values when a time grain is used to ensure that the calculation window always has the correct number of lagged values applied.

DB migration benchmark results:
```
Current: 0.42 s
10+: 0.33 s
100+: 0.30 s
1000+: 0.30 s
10000+: 0.42 s
```


### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF
Before:

![image](https://user-images.githubusercontent.com/15073128/144034565-8987a193-fee1-4f2a-a774-b9a4e3f14f6b.png)

After:

![image](https://user-images.githubusercontent.com/15073128/144034282-1fcab02a-ddf8-4e89-beb0-0cd064c40308.png)

### TESTING INSTRUCTIONS
Big Number and Big Number with Trendline core functionalities should work like before.
Timestamp metrics should be formatted automatically.

Steps to reproduce bug shown on screenshots:
1. Create a BigNumber or Big Number with Trendline chart using SQLite or Druid datasource
2. Create a metric that should return a date (like `MAX(temporal_column)`)
3. Verify that Big Number and Trendline are rendered properly, and that big number and trendline's tooltip are formatted as a date

### ADDITIONAL INFORMATION



CC @junlincc @jinghua-qa
